lib/collection.js

```diff
@@ -0,0 +1,290 @@
+var Events = require('./events'),
+    Collection,
+    ARRAY_PROXY_METHODS = ['forEach', 'some', 'every', 'filter', 'map', 'reduce'];
+
+function clone(obj) {
+    return Object.keys(obj).reduce(function (result, key) {
+        result[key] = obj[key];
+        return result;
+    }, {});
+}
+
+/**
+ * @class Collection
+ * @extends Events
+ */
+Collection = Events.inherit(/** @lends Collection.prototype*/{
+
+    /**
+     * @type {Function}
+     */
+    modelType: null,
+
+    /**
+     * @param {Array.<(Model|Object)>} [data]
+     */
+    __constructor: function (data) {
+        this.__base.apply(this, arguments);
+
+        /**
+         * @type {Array.<Model>}
+         */
+        this._models = [];
+
+        /**
+         * @type {Object.<string, Model>}
+         */
+        this._modelsIdsMap = {};
+
+        Object.defineProperty(this, 'length', {
+            get: function () {
+                return this._models.length;
+            }
+        });
+
+        data = data || [];
+        this.set(data);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    isChanged: function (branch) {
+        return this._models.some(function (model) {
+            return model.isChanged(branch);
+        });
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    commit: function (branch) {
+        return this._models.reduce(function (changed,  model) {
+            return model.commit(branch) || changed;
+        }, false);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Collection}
+     */
+    revert: function (branch) {
+        this._models.forEach(function (model) {
+            model.revert(branch);
+        });
+        return this;
+    },
+
+    /**
+     * @returns {Array.<Object>}
+     */
+    toJSON: function () {
+        return this._models.map(function (model) {
+            return model.toJSON();
+        });
+    },
+
+    /**
+     * @param {Number} index
+     * @returns {Model}
+     */
+    at: function (index) {
+        return this._models[index];
+    },
+
+    /**
+     * @param {*} id
+     * @returns {?Model}
+     */
+    get: function (id) {
+        return this._modelsIdsMap[id] || null;
```

Comment:
why null is returned here on missing model and undefined is returned in "at" method in similar situation?